### PR TITLE
Adding option to tag objects on creation

### PR DESCRIPTION
**Pull request by seandc**

### CODE REVIEW

File: lib/uber-s3/operation/object/tagging.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module UberS3::Operation::Object
```
Comment (seandc) [inline lib/uber-s3/operation/object/tagging.rb:1]: This is the pattern they used for adding options to the request. It is weird but I followed it for consistency reasons.

File: lib/uber-s3/operation/object/tagging.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      base.instance_eval do
+        attr_accessor :tagging
+      end
+    end
+
+    module ClassMethods
+    end
+    
+    module InstanceMethods
+      def tagging=(tags)
```
Comment (seandc) [inline lib/uber-s3/operation/object/tagging.rb:17]: Not sure if this validation is needed.

File: Gemfile.lock [hunk 3]
```diff
@@ -34,8 +34,8 @@ GEM
     diff-lcs (1.2.5)
     em-socksify (0.2.0)
       eventmachine (>= 1.0.0.beta.4)
-    eventmachine (1.0.3)
-    eventmachine (1.0.3-java)
+    eventmachine (1.0.9.1)
+    eventmachine (1.0.9.1-java)
```
Comment (seandc) [inline Gemfile.lock:38]: I could not build eventmachine on my linux box without bumping the patch version.

File: lib/uber-s3/operation/object/tagging.rb (hunk 1 above)
Comment (maxl0rd) [reply lib/uber-s3/operation/object/tagging.rb:1]: yea rails mixin style

**Review by maxl0rd — APPROVED** [review]
lgtm

File: lib/uber-s3/object.rb [hunk 4]
```diff
@@ -89,6 +90,14 @@ def save
       if !meta.nil? && !meta.empty?
         meta.each {|k,v| headers["x-amz-meta-#{k}"] = v.to_s }
       end
+
+      # Tagging
+      if !tagging.nil? && !tagging.empty?
+        # Turn the tags hash into a query string
+        tagging_query_string = tagging.map { |k, v| "#{k}=#{v}" }.join('&')
```
Comment (adamvduke) [inline lib/uber-s3/object.rb:97]: is there any chance that `k` or `v` will have characters that need to be escaped?

**Review by jeremydurham — CHANGES_REQUESTED** [review]
Bump version please

**Pull request closed without merging**